src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -331,10 +334,11 @@ substXinS a_fresh name payload statements
 --  * Remove some other useless code:
 --      statements that do not update accumulators or return a value are silly.
 --
-thresher :: (Ord n, Eq p) => a -> Statement a n p -> FixT (Fresh n) (Statement a n p)
+thresher :: (Hashable n, Eq n, Eq p) => a -> Statement a n p -> FixT (Fresh n) (Statement a n p)
 thresher a_fresh statements
- = transformUDStmt trans emptyExpEnv statements
+ = fmap (reannotS fst) $ transformUDStmt trans emptyExpEnv $ freevarsStmt statements
```

Comment:
actually, I guess freevars could probably be replaced by allvars in nestBlocks, since it's just going to be a bit more pessimistic in what it needs to rename (can't remember if you tried that)